[IV-21-22] Objetivo 2

agr8 opened the pull request:
- Se ha organizado el repositorio siguiendo las buenas prácticas de python, creando para ello el directorio '_**inmuventa**_'.

- Se ha creado la estructura de datos que representa un inmueble en el fichero '**_inmueble.py_**'

Esto avanza la implementación del primer Milestone y cierra los issues #18 y #20 .

agr8 committed:
Creo el fichero iv.yaml

agr8 committed:
Añado lenguaje y entidad al fichero iv.yaml

agr8 committed:
Añado fichero inmueble.py

He creado una dataclass para representar un inmueble. Tambien he creado una dataclass para poder representar la localizacion de un inmueble: los argumentos de su metodo init que tienen como valor por defecto 'None' son aquellos que se han descrito como 'opcionales'. This closes #18, closes #20

agr8 committed:
Corrijo fichero iv.yaml

agr8 commented:
Ya han pasado correctamente los tests. @ismaelmontesinos, revisa el PR cuando puedas por si tengo que modificar algo para cumplir con el objetivo. :pray:

JJ commented on inmuventa/inmueble.py:
```diff
@@ -0,0 +1,40 @@
+        self.nombre_via = nombre_via
+        self.bloque = numero_bloque
+        self.planta = planta
+        self.numero_puerta = numero_puerta
```
¿Todo esto influye en la lógica de negocio? Si es simplemente pare identificar al inmueble, en un PMV no hace falta.

JJ commented on inmuventa/inmueble.py:
```diff
@@ -0,0 +1,40 @@
+        self.puerta = puerta
+
+@dataclass
+class Inmueble:
```
Si no va a variar, por favor considera usar "Frozen"

JJ reviewed (changes requested):
Aparte de los comentarios en línea, mira las buenas prácticas en mensajes de commit.

agr8 committed:
Modifico la clase Inmueble para que sea inmutable.

Esto se debe a que una instancia de dicha clase no variará. Para ello, he utilizado 'Frozen', estableciendo (frozen=True).

agr8 committed:
Elimino atributos opcionales

He eliminado todos los atributos opcionales ya que no influían en la lógica de negocio. This fixes #18, closes #20

agr8 committed:
Hago inmutable la clase Localizacion

Modifico la dataclass para hacerla inmutable estableciendo (frozen=True).
This fixes #18, fixes #20

agr8 commented:
@JJ, cambios realizados también en la clase Localizacion. He eliminado todos los atributos opcionales y he utilizado _(frozen=True)_ para hacer inmutables ambas clases. También he intentado seguir las buenas prácticas en los nuevos commits. De ahora en adelante, tendré más cuidado para cumplir con las buenas prácticas :sweat_smile:

JJ reviewed (approved):
Se han intentado seguir las buenas prácticas, pero rara vez un commit cerrará dos issues. Puedes referirte a un PR tal como este si estás tratando de resolver lo que se comenta en una revisión.
Rogaría a @ismaelmontesinos que fuera un poco más diligente revisando y haciendo sugerencias. Una vez que ya lo haya aprobado yo, @agr8 no tiene por qué hacer ningún cambio más y tendrás que usar lo que haya para tu siguiente objetivo.

ismaelmontesinos commented:
Por mi parte está todo perfecto tal y como hemos comentado tanto en el issue (#18) como por telegram y en presencial, por lo tanto por mi parte estaría listo para el merge.

JJ reviewed (approved):
No es buena práctica cerrar dos issues con el mismo commit, por lo demás correcto.

agr8 commented:
Lo tengo en cuenta para los siguientes objetivos, gracias :smile:
